packages/dev-server-rollup/src/rollupAdapter.ts

```diff
@@ -215,8 +217,57 @@ export function rollupAdapter(
 
         return;
       }
+
+      if (context.response.is('html')) {
+        const documentAst = parseHtml(context.body);
+        const inlineModuleNodes = queryAll(
+          documentAst,
+          predicates.AND(
+            predicates.hasTagName('script'),
+            predicates.hasAttrValue('type', 'module'),
+            predicates.NOT(predicates.hasAttr('src')),
+          ),
+        );
+
+        const filePath = resolveFilePath(rootDir, context.path);
+
+        for (const node of inlineModuleNodes) {
+          const code = getTextContent(node);
+
+          const rollupPluginContext = createRollupPluginContextAdapter(
+            rollupPluginContexts.transformPluginContext,
+            wdsPlugin,
+            config,
+            fileWatcher,
+            context,
+          );
+
+          const result = await rollupPlugin.transform?.call(
+            rollupPluginContext as TransformPluginContext,
+            code,
+            filePath,
+          );
+
+          let transformedCode: string | undefined = undefined;
+          if (typeof result === 'string') {
+            transformedCode = result;
+          }
+
+          if (typeof result === 'object' && typeof result?.code === 'string') {
+            transformedCode = result.code;
+          }
+
+          if (transformedCode) {
+            transformedFiles.add(context.path);
+            return transformedCode;
```

Comment:
@LarsDenBakker Im not sure how I should go about this here, returning the transformedCode for a single file makes sense, but here I'm wanting to do the transform on potentially multiple inline scripts, so if I return here, if theres a 2nd inline script, it wont get the transform, because the function has already returned

Reply:
If you return here the HTML file becomes a JS file :D 

I think you need to replace the content of the HTML script node with the transformed code text. And then send back the updated HTML.

Reply:
> If you return here the HTML file becomes a JS file :D

Its a feature!

Reply:
@LarsDenBakker Alright, I *think* this does what it should be doing now, could you take another look?